src/main/java/com/faforever/client/main/MainController.java

```diff
@@ -476,7 +476,11 @@ private void enterLoggedInState() {
     getMainScene().setContent(mainRoot);
     getMainScene().setMoveControl(mainRoot);
 
-    clientUpdateService.checkForRegularUpdateInBackground();
+    if (preferencesService.getPreferences().isPrereleaseCheckEnabled()) {
+      clientUpdateService.checkForRegularUpdateInBackground();
+    } else {
+      clientUpdateService.checkForBetaUpdateInBackground();
```

Comment:
Isn't one method with a boolean parameter a better choice? Why create 2 methods?
And why is the preference read in a UI class, shouldn't the clientUpdateService just read the property itself?

Reply:
Well this is after Logging in. But yeah we could have an event here...